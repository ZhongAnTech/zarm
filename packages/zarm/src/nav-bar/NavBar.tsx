import React, { forwardRef } from 'react';
import type { HTMLAttributes } from 'react';
import classnames from 'classnames';
import { ConfigContext } from '../n-config-provider';
import type { BaseNavBarProps } from './interface';

export type NavBarProps = BaseNavBarProps & Omit<HTMLAttributes<HTMLDivElement>, 'title'>;

const NavBar = forwardRef<HTMLDivElement, NavBarProps>((props, ref) => {
  const { className, title, left, right, ...restProps } = props;

  const { prefixCls: globalPrefixCls } = React.useContext(ConfigContext);
  const prefixCls = `${globalPrefixCls}-nav-bar`;

  const cls = classnames(prefixCls, className);
  const titleCls = `${prefixCls}__title`;
  const sideCls = `${prefixCls}__side`;
  const leftCls = `${sideCls} ${prefixCls}__side--left`;
  const rightCls = `${sideCls} ${prefixCls}__side--right`;

  return (
    <div ref={ref} className={cls} {...restProps}>
      <div className={leftCls}>{left}</div>
      <div className={titleCls}>{title}</div>
      <div className={rightCls}>{right}</div>
    </div>
  );
});

NavBar.displayName = 'NavBar';

NavBar.defaultProps = {};

export default NavBar;
