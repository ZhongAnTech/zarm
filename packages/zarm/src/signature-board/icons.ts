export const DEL_ICON =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAMAAABHPGVmAAAAflBMVEUAAABRUVFRUVFRUVFPT09PT09QUFBQUFBKSkpRUVFPT09QUFBRUVFSUlJRUVFRUVFRUVFSUlJRUVFQUFBRUVFRUVFRUVFRUVFQUFBRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFRUVFSUlJPT09QUFBRUVFVVVVYWFhTU1ORL5V1AAAAJnRSTlMAA/r8BSqAuQ7xR0a3fULqqBPzCuXMXVIZo5qFbzbfwXho1zAikA+znhcAAAKfSURBVGje7ZrZjqMwEEXLS8ja7EtCtl5d9v//4IB7JAgjdWFgeqYln0enrg8FETg44PlxMMlmltAKh8LpDhk+nQ7rrzkF+2yypclVhdFIos2tnN7MBxolxmCwhomctBBqFELoekorDC6a2wk4gbBV+DLFkuTGxlEToBKNxOSJs0LCBVuHSeOA4MRNa8FLE3LlbiXpG5BcI3u+7uBMlps2eQZJAkF7PCbPnCWbrWowO/ocMKiMathuJkrUHhgt2at5kt0PlLAHOolkBBJ2vyWDOeh7tfyUhDCC8FMiyUmz1SPX1CbL9xXJe2lL0+tgPBs6niM+QFn4F1C1UfDYVqmNWhyjK2A9SYx8eQnH+Jsk33e6ugu/OFFAfIWXIHNYOC25bGOLA/8eJv9Y3kopYcwQcxMRQ8TnNPuP82oQKetL0h9ikFzqcuA4vlzexrZwRtSD526sUb8mDxO+aq3jh9w1QjQj13jHVHGORT++Q8G5DkD2jkRzLkzYn/LeDJkoGdVIqFqiTW+hF2iuBB6gxxqF4vqp80Jml1/KeknJzkrSUZKgJ9mkNrhzkGwdO+kWUV7iJV7iJV7iJV7iJV7iJV7iJX9RcraSNfEjaKbkBbniWAPrqk52qJwt6UhyrdBc+5IQUelbVzRbwuDtEBUVPFAV0WEFbDFJG9vAADvEFumks3TZ/tB8CY2XeMm3SnadhHwVlSaukmNqg+EoyTtXDSYE6QSExkquVkKR3YxQwhSJYyOF3cbdHoHG3sXtZu6tDhyoc5vqNhqoi2IPSRjULthQ91aStMSatwHBHRCqgesT6ehex2ITckZwLDbjX24ndzTCGYPF0eUPIvI5Nxqd0CZ/ln0HbYGkOsdrB+Jz1T0px2to5oeYZG7l/8OmjGcsvwB+KxB3VpOtpgAAAABJRU5ErkJggg==';

export const UNDO_ICON =
  'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAGQAAABkCAMAAABHPGVmAAAAhFBMVEUAAAAsLCwnJycsLCwsLCwlJSUsLCwyMjIsLCwsLCwsLCwsLCwsLCwrKyssLCwsLCwsLCwsLCwsLCwsLCwsLCwpKSksLCwrKyssLCwrKyssLCwsLCwsLCwrKyssLCwsLCwsLCwsLCwrKystLS0sLCwrKyssLCwqKiosLCwsLCwsLCwuLi7xqpc4AAAAKnRSTlMA/AOcYw7yBfo5dvZtSdOjXePakk8J6ooyEe+0rZgifkM+KhzJtykYysZAy40uAAAB+UlEQVRo3u3Y247aQAyAYTscEhJCQgiQcNou7KG13//96tHOwEVVElf1Rav579En4yA5QCwWi8Vi/1kJ/NI/anyk1lAKXXG2HUeMRUG8sVOCgWiqiJEREjnFZ2ZYKg/DKWeQbA3iHCRjYwmSuZFE4/fG690gZ6TJs/6CkdUwmEBq49vDQGo2kyft87r81DNJMEI8EB3e804+pzPQGz4ciIgZq4uCSaB0hi6BeL7vITUyQsI0V0jGGa039GVc1F4ZmqOgAnWRpPrV1qyPEB9KOaxcNtV5ouv0Y0fiBGV3GVRS+IP6tkIOCq+2w0tRB67uyPft5zYnh5P2XkFa92CTKMug8EsYxYDZM/pRZmCTIMkqKK3hKOXjeAqIgfLO+PV9bcEoQXKHuDrD72tBvtIKkT4Lv5TaENmuPZJbIg2hR1Kwqp/7SaaGi79m9FVriNThEV4YIhOPzG9gkxi3nV/JEYwSpGI03nsKLfmKiw0ixseB/CAnGyMBeN3dL5ZOg2hOiZy8gVzZDNKXbxwMyq4aJK2nY8r3px0xkk/5aKVrHhlRMDLtqZKuKFNe3BmfQNIhqH1DqQASSwTFeHGGDRKIpnWGCYIuYm6WWzUhSMM0Jpbmx+nMjaFH3rLDfLDD+vuxmt7AE+pmo9om5v9Qh/eTWCwWi8VisdizfgJaxKDFfwfcTwAAAABJRU5ErkJggg==';
