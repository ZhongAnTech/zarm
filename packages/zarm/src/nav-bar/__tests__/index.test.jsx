import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ArrowLeft, ArrowRight } from '@zarm-design/icons';
import NavBar from '../index';

describe('NavBar', () => {
  const props = {};

  it('renders correctly', () => {
    const wrapper = render(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with title', () => {
    props.title = '测试标题';
    const wrapper = render(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with left content', () => {
    props.left = <ArrowLeft />;
    const wrapper = render(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with right content', () => {
    props.right = <ArrowRight />;
    const wrapper = render(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders correctly with multiple icons', () => {
    const infoRoundSvg = (
      <i
        style={{
          fontSize: '14px',
          color: 'var(--theme-danger)',
          width: '1em',
          height: '1em',
          lineHeight: '1em',
          display: 'inline-block',
        }}
      >
        <svg
          width="1em"
          height="1em"
          viewBox="0 0 37 37"
          fill="currentColor"
          stroke="currentColor"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g stroke="none" fill="none" fillRule="evenodd">
            <path
              d="M35 18.5C35 9.387 27.613 2 18.5 2S2 9.387 2 18.5 9.387 35 18.5 35 35 27.613 35 18.5z"
              stroke="currentColor"
              strokeWidth="2.6"
              strokeLinecap="round"
              strokeLinejoin="round"
            />
            <g fill="currentColor">
              <path d="M18.103 13.793c-.61 0-1.103.494-1.103 1.104v12a1.103 1.103 0 1 0 2.207 0v-12c0-.61-.493-1.104-1.104-1.104M18.103 9.791c-.143 0-.286.033-.419.088A1.108 1.108 0 0 0 17 10.894c0 .298.12.574.32.784.11.1.22.187.364.243.408.165.894.066 1.203-.243.199-.21.32-.486.32-.784 0-.143-.033-.286-.088-.419a1.09 1.09 0 0 0-.232-.353 1.122 1.122 0 0 0-.784-.331" />
            </g>
          </g>
        </svg>
      </i>
    );
    props.left = (
      <div>
        <ArrowLeft />
      </div>
    );
    props.right = (
      <div>
        <ArrowRight />
        {infoRoundSvg}
      </div>
    );
    const wrapper = render(<NavBar {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
