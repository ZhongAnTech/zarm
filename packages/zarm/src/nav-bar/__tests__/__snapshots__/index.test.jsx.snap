// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  />
  <div
    class="za-nav-bar__title"
  />
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  />
</div>
`;

exports[`NavBar renders correctly with left content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  >
    <i
      class="za-icon za-icon--default za-icon--md"
      theme="default"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1000 1000"
        width="1em"
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
  <div
    class="za-nav-bar__title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  />
</div>
`;

exports[`NavBar renders correctly with multiple icons 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  >
    <div>
      <i
        class="za-icon za-icon--default za-icon--md"
        theme="default"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1000 1000"
          width="1em"
        >
          <path
            d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
            fill-rule="evenodd"
          />
        </svg>
      </i>
    </div>
  </div>
  <div
    class="za-nav-bar__title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  >
    <div>
      <i
        class="za-icon za-icon--default za-icon--md"
        theme="default"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 1000 1000"
          width="1em"
        >
          <path
            d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
            fill-rule="evenodd"
          />
        </svg>
      </i>
      <i
        style="font-size:14px;color:var(--theme-danger);width:1em;height:1em;line-height:1em;display:inline-block"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          viewBox="0 0 37 37"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
          >
            <path
              d="M35 18.5C35 9.387 27.613 2 18.5 2S2 9.387 2 18.5 9.387 35 18.5 35 35 27.613 35 18.5z"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2.6"
            />
            <g
              fill="currentColor"
            >
              <path
                d="M18.103 13.793c-.61 0-1.103.494-1.103 1.104v12a1.103 1.103 0 1 0 2.207 0v-12c0-.61-.493-1.104-1.104-1.104M18.103 9.791c-.143 0-.286.033-.419.088A1.108 1.108 0 0 0 17 10.894c0 .298.12.574.32.784.11.1.22.187.364.243.408.165.894.066 1.203-.243.199-.21.32-.486.32-.784 0-.143-.033-.286-.088-.419a1.09 1.09 0 0 0-.232-.353 1.122 1.122 0 0 0-.784-.331"
              />
            </g>
          </g>
        </svg>
      </i>
    </div>
  </div>
</div>
`;

exports[`NavBar renders correctly with right content 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  >
    <i
      class="za-icon za-icon--default za-icon--md"
      theme="default"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1000 1000"
        width="1em"
      >
        <path
          d="M296.114 508.035c-3.22-13.597.473-28.499 11.079-39.105l333.912-333.912c16.271-16.272 42.653-16.272 58.925 0s16.272 42.654 0 58.926L395.504 498.47l304.574 304.574c16.272 16.272 16.272 42.654 0 58.926s-42.654 16.272-58.926 0L307.241 528.058a41.472 41.472 0 01-11.127-20.023z"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
  <div
    class="za-nav-bar__title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  >
    <i
      class="za-icon za-icon--default za-icon--md"
      theme="default"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="0 0 1000 1000"
        width="1em"
      >
        <path
          d="M711.156 508.035a41.472 41.472 0 01-11.126 20.023L366.12 861.97c-16.272 16.272-42.654 16.272-58.926 0s-16.272-42.654 0-58.926L611.767 498.47 307.241 193.944c-16.272-16.272-16.272-42.654 0-58.926 16.271-16.272 42.653-16.272 58.925 0L700.078 468.93c10.606 10.606 14.299 25.508 11.078 39.105z"
          fill-rule="evenodd"
        />
      </svg>
    </i>
  </div>
</div>
`;

exports[`NavBar renders correctly with title 1`] = `
<div
  class="za-nav-bar"
>
  <div
    class="za-nav-bar__side za-nav-bar__side--left"
  />
  <div
    class="za-nav-bar__title"
  >
    测试标题
  </div>
  <div
    class="za-nav-bar__side za-nav-bar__side--right"
  />
</div>
`;
